private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -391,22 +546,6 @@ func (p *moduleDataStore) putModuleData(
 	return storage.PutPath(ctx, moduleCacheBucket, externalModuleDataFileName, data)
```

Comment:
We should write this file atomically - that way if we're ever killed (i.e. kill -9) we never have the chance to read an invalid module.yaml file.